### ISSUE TITLE
printing >= 4 times with the esp_log.h library conflicts with Serial.print() (IDFGH-13196)

### ISSUE THREAD

**Issue by shyney7** [espressif/arduino-esp32#9985]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

framework-arduinoespressif32 @ 3.20016.0 (2.0.16)

### Espressif SoC revision.

ESP32-S3 (QFN56) (revision v0.1)

### Operating System used.

Windows

### How did you build your project?

VS Code IDE

### If you are using Windows, please specify command line type.

PowerShell

### Development Kit.

WEMOS LOLIN S3 PRO

### Power Supply used.

USB

### What is the expected behavior?

Using the esp_log.h library sould not conflict with Serial.print() function.

### What is the actual behavior?

When printing e.g. with the ESP_LOGD() macro >= 4 times and then using Serial.print() to output something to the serial monitor no output is visible. The same problem was already reported 4 years ago here: https://github.com/espressif/arduino-esp32/issues/4261 . This issue was closed although the problem seems to still exist. It doesnt matter wich print function you use (Serial.printf() etc.) the output is not visible besides the output from esp_log.h

### Steps to reproduce.

1. Include esp_log.h
2. Use any print macro e.g. ESP_LOGD() equal or more than 4 times subsequently
3. Use any Serial print function like Serial.print() or Serial.printf() right after
4. Observe that there is no output to the serial monitor besides the ESP_LOGD() output.

### More Information.

It seems even printing only once with a macro causes this issue.
Here is a minimal example:
```cpp
#include <Arduino.h>
#include <esp_log.h>

void setup() {
  Serial.begin(9600);
}

void loop() {

  ESP_LOGD("Test_Debug", "Test123");
  #if CORE_DEBUG_LEVEL >= 4
    Serial.println("Serialprint Output");
  #endif
}
```

Output:
```shell
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x15 (USB_UART_CHIP_RESET),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x4202225a
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3808,len:0x4bc
load:0x403c9700,len:0xbd8
load:0x403cc700,len:0x2a0c
entry 0x403c98d0
[    95][I][esp32-hal-psram.c:96] psramInit(): PSRAM enabled
[   110][D][main.cpp:27] loop(): [Test_Debug] Test123
[   167][D][main.cpp:27] loop(): [Test_Debug] Test123
[   245][D][main.cpp:27] loop(): [Test_Debug] Test123
[   323][D][main.cpp:27] loop(): [Test_Debug] Test123
[   402][D][main.cpp:27] loop(): [Test_Debug] Test123
[   480][D][main.cpp:27] loop(): [Test_Debug] Test123
[   558][D][main.cpp:27] loop(): [Test_Debug] Test123
[   636][D][main.cpp:27] loop(): [Test_Debug] Test123
[   714][D][main.cpp:27] loop(): [Test_Debug] Test123
[   792][D][main.cpp:27] loop(): [Test_Debug] Test123
[   870][D][main.cpp:27] loop(): [Test_Debug] Test123
[   948][D][main.cpp:27] loop(): [Test_Debug] Test123
[  1027][D][main.cpp:27] loop(): [Test_Debug] Test123
[  1105][D][main.cpp:27] loop(): [Test_Debug] Test123
[  1183][D][main.cpp:27] loop(): [Test_Debug] Test123
[  1261][D][main.cpp:27] loop(): [Test_Debug] Test123
[  1339][D][main.cpp:27] loop(): [Test_Debug] Test123
[  1417][D][main.cpp:27] loop(): [Test_Debug] Test123
[  1495][D][main.cpp:27] loop(): [Test_Debug] Test123
[  1574][D][main.cpp:27] loop(): [Test_Debug] Test123
[  1652][D][main.cpp:27] loop(): [Test_Debug] Test123
[  1730][D][main.cpp:27] loop(): [Test_Debug] Test123
[  1808][D][main.cpp:27] loop(): [Test_Debug] Test123
[  1886][D][main.cpp:27] loop(): [Test_Debug] Test123
[  1964][D][main.cpp:27] loop(): [Test_Debug] Test123
[  2042][D][main.cpp:27] loop(): [Test_Debug] Test123
[  2120][D][main.cpp:27] loop(): [Test_Debug] Test123
[  2199][D][main.cpp:27] loop(): [Test_Debug] Test123
[  2277][D][main.cpp:27] loop(): [Test_Debug] Test123
[  2355][D][main.cpp:27] loop(): [Test_Debug] Test123
```

**Comment by me-no-dev** [comment]
Please post minimal example code that will reproduce the problem. there is no reason IDF's LOG to interfere with Serial output.

**Comment by shyney7** [comment]
I've uploaded my code with the Arduino IDE instead of platformio and its working now. I dont know what caused this issue in the first place. It was so far that no matter if I used the esp_log library or not the Serial.print() statements were not output to the serial monitor. I don't know what caused this to happen but it doesn't seem to be the esp_log library. After I uploaded the sketch with the Arduino IDE and uploaded my code via full clean back in platformio it worked.
